artifact-manifest.md

```diff
@@ -0,0 +1,551 @@
+{
+  "schemaVersion": 1,
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType": "application/vnd.cncf.notary.v2+json",
+  "config": {
+    "mediaType": "application/vnd.cncf.notary.config.v2+json",
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    "size": 102
+  },
+  "blobs": [
+    {
+      "mediaType": "application/tar",
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+      "size": 32654
+    }
+  ],
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+      "size": 16724
+    }
+  ],
+  "annotations": {
+    "org.cncf.notary.v2.signature.subject": "docker.io"
+  }
+}
+```
+
+![Notary v2 signature](./media/sbom-document.svg)
+
+**An SBoM of the `wordpress:5` image:**
+
+```json
+{
+  "schemaVersion": 1,
+  "mediaType": "application/vnd.oci.artifact.manifest.v1+json",
+  "artifactType": "application/vnd.example.sbom.v1+json",
+  "config": {
+    "mediaType": "application/vnd.example.sbom.config.v1+json",
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9bce1f432a537bc7",
+    "size": 102
+  },
+  "blobs": [
+    {
+      "mediaType": "application/tar",
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28f2f9d09af107ee8f0",
+      "size": 32654
+    }
+  ],
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.manifest.v1+json",
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1905b15da2eb4a4c6b",
+      "size": 16724
+    }
+  ],
+  "annotations": {
+    "example.sbom.author": "docker.io"
+  }
+}
+```
+
+Neither the signature, nor the SBoM, have value unto themselves. A signature and SBoM extend a given artifact. They're considered as dependent upon another artifact. The `manifests` collection references the `wordpress:5` digest, creating the dependency.
+
+![wordpress image with layers](media/wordpress-image-layers-sig-sbom.svg)
+
+The Notary v2 signature and SBoM reference the `wordpress:5` image through the `manifests` collection. The `wordpress:5` image is represented as an oci-image, and requires no changes to its manifest to support the enhancements. The directionality of the references indicates the references in the `wordpress:5` image & the Notary v2 and SBoM manifest references to existing content.
+
+Each OCI artifact manifest is independent, supporting multiple signatures, SBoMs and other artifacts to be added after the image was persisted. While an image knows of it's layers, and a Notary v2 signature knows of its config and blob, the image manifest does not know about the Notary v2 signature, nor the SBoM. The Notary v2 signature declares a dependency to the artifact it's signing. Upon upload completion, the registry would persist a reference between the `wordpress:5` image and the Notary v2 signature.
```

Comment:
> the registry would persist a reference between the `wordpress:5` image and the Notary v2 signature

Do we want to provide any guidance here on how to do so?

Reply:
Are you asking about how the registry would store the persistence, as in the CNCF distribution instance? If so, we'll have an updated PR, similar to the one Aviral did here: https://github.com/notaryproject/distribution/pull/2

Reply:
Plesae see the `artifact-manifest-spec.md` in this PR for details.

And the prototype-2 fork of distribution for an experimental implementation:
- implementation: https://github.com/notaryproject/distribution/tree/prototype-2
- docs of persisence: https://github.com/notaryproject/distribution/pull/9
- oras cli experimental validation: https://github.com/deislabs/oras/blob/prototype-2/docs/artifact-manifest.md